apis/monitoring/v1beta1/monitoringdashboard_types.go

```diff
@@ -0,0 +1,898 @@
+	/* Immutable. The Project that this resource belongs to. */
+	ProjectRef v1alpha1.ResourceRef `json:"projectRef"`
+
+	/* Immutable. Optional. The name of the resource. Used for creation and acquisition. When unset, the value of `metadata.name` is used as the default. */
+	// +optional
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Required. The mutable, human-readable name.
+	//
+	// +required
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// Content is arranged with a basic layout that re-flows a simple list of
+	//  informational elements like widgets or tiles.
+	GridLayout *GridLayout `json:"gridLayout,omitempty"`
+
+	// The content is arranged as a grid of tiles, with each content widget
+	//  occupying one or more grid blocks.
+	MosaicLayout *MosaicLayout `json:"mosaicLayout,omitempty"`
+
+	// The content is divided into equally spaced rows and the widgets are
+	//  arranged horizontally.
+	RowLayout *RowLayout `json:"rowLayout,omitempty"`
+
+	// The content is divided into equally spaced columns and the widgets are
+	//  arranged vertically.
+	ColumnLayout *ColumnLayout `json:"columnLayout,omitempty"`
+
+	/*NOTYET
+	// Filters to reduce the amount of data charted based on the filter criteria.
+	DashboardFilters []DashboardFilter `json:"dashboardFilters,omitempty"`
+
+	// Labels applied to the dashboard
+	Labels []Dashboard_LabelsEntry `json:"labels,omitempty"`
+	*/
+}
+
+// +kcc:proto=google.monitoring.dashboard.v1.Dashboard
+type MonitoringDashboardStatus struct {
+	/* Conditions represent the latest available observations of the
+	   MonitoringDashboard's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+	/* \`etag\` is used for optimistic concurrency control as a way to help prevent simultaneous updates of a policy from overwriting each other. An \`etag\` is returned in the response to \`GetDashboard\`, and users are expected to put that etag in the request to \`UpdateDashboard\` to ensure that their change will be applied to the same version of the Dashboard configuration. The field should not be passed during dashboard creation. */
+	// +optional
+	Etag *string `json:"etag,omitempty"`
+
+	/* ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource. */
+	// +optional
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories=gcp,shortName=gcpmonitoringdashboard;gcpmonitoringdashboards
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels="cnrm.cloud.google.com/dcl2crd=true";"cnrm.cloud.google.com/managed-by-kcc=true";"cnrm.cloud.google.com/stability-level=stable";"cnrm.cloud.google.com/system=true"
```

Comment:
Note that because we still have the dcl2crd annotation (for now) we don't "break" the existing reconciler (yet).